authenticate/authenticate.go

```diff
@@ -28,15 +25,6 @@ func ValidateOptions(o *config.Options) error {
 	if _, err := cryptutil.NewAEADCipherFromBase64(o.CookieSecret); err != nil {
 		return fmt.Errorf("authenticate: 'COOKIE_SECRET' invalid %w", err)
 	}
-	if o.Provider == "" {
```

Comment:
To confirm, this means that the authenticate service can now run without a global idp credential set, right?

Reply:
Yeah -- it appears so. This get's wonky fast, but I don't think it breaks anything. For example, I could have a configuration like follows. 

```yaml
idp_provider: google
#notice, no other client idp configuration details for global?

policy:
  - from: https://httpbin.localhost.pomerium.io
    to: https://httpbin.org
    idp_client_id: bar.apps.googleusercontent.com
    idp_client_secret: foo
    pass_identity_headers: true
    allowed_users:
      - bdd@pomerium.com
      - bobbydesimone@gmail.com
  - from: https://httpbin2.localhost.pomerium.io
    to: https://httpbin.org
    pass_identity_headers: true
    allowed_users:
      - bdd@pomerium.com
      - bobbydesimone@gmail.com
```

httpbin works but httpbin2 hits a redirect flow error. I'm not sure this is worth guarding against. 

<img width="1336" alt="Screen Shot 2022-02-09 at 4 18 32 PM" src="https://user-images.githubusercontent.com/1544881/153312937-e66f3e55-404a-4638-8748-720cf66dc374.png">

Reply:
I don't know the right requirements, but I can re-add the restriction

Reply:
I think this is fine behavior.